tutorials/haskell/dotenv-hs/tutorial.md

```diff
@@ -0,0 +1,284 @@
+---
+title: Setting up your environment with dotenv
+published: not-yet
+ghc: 8.0.2
+lts: 8.4
+libraries: dotenv-0.4.0.0
+language: haskell
+author-name: Cristhian Motoche
+description: Check Out the two working modes of dotenv to set up an
+environment before the execution of an application.
+---
+# Introduction
+## What is dotenv?
+[Dotenv][dotenv] is a Haskell library and an Standalone tool that enables to set
+up the configuration of an environment before the execution of an application.
+As it is suggested in [The Twelve Factor App](https://12factor.net/config)
+the configuration should be separated from the code. A good way to do that is
+by loading **environment variables** (env vars) and `dotenv` is the perfect tool
+for that, because it loads all the needed env vars for an application to run.
+
+## Why it is important?
+The configuration should be separated from the code for one simple reason: the
+code should be the same in every environment (development, testing, staging,
+quality assurance, production, etc.) and configuration is the opposite, it
+changes in every environment.
+
+Loading env vars is a good way to set up the configuration of an application,
+because they can change easily everytime the environment changes and without
+changing code.
+
+## How it works?
+Dotenv reads the needed application's env vars defined in a *dotenv example file*
+(necesary env vars). If the necesary env vars are defined in *the environment*,
+they stay the same. If they are not there, then dotenv checks if they defined
+in a *dotenv file* and sets all the env vars that are there. Otherwise, it will
+fail, because the necesary env vars are missing. In this way, there is a certanty
+that the configuration of the application is set up before its execution.
+
+# Examples
+The example will be a simple application that prints the content of a table in a
```

Comment:
I thought about this example, because it shows a common use of `dotenv` in a real world application. Perhaps, some of our readers haven't heard about something like `dotenv` before, this example could help them to understand the power of it.

WDYT?